tests/test_protocol_hardening.py

```
@@ -162,3 +161,71 @@ async def test_duplicate_tx_and_block_detection(self):
+        })
+        self.assertIsNone(result)
+
+    async def test_valid_chain_request_is_accepted(self):
+        handler = self._make_handler()
+        # A well-formed request should not be dropped (result is None only on error return).
+        try:
+            await self._call(handler, {
+                "type": "chain_request",
+                "data": {"start_index": 0, "limit": 10},
+            })
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)